vignettes/data-transform-as-shiny-module.Rmd

```diff
@@ -0,0 +1,172 @@
+---
+title: "Data Transform as shiny Module"
+author: "NEST CoreDev"
+output: 
+  rmarkdown::html_vignette:
+    toc: true
+vignette: >
+  %\VignetteIndexEntry{Data Transform as shiny Module}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+## Introduction
+
+The input data supplied by the app developer in the `data` argument of the `init` function is passed through the filter panel and the filtered data is supplied to the all modules.
+It is important to note that the filter panel only performs one kind of computation on the datasets, it filters the data based on the user interaction.
+However, there are use cases where we need to provide a more customized transformation options for the app user.
+Historically, customised data transformations was achieved in the teal apps using the encoding panel, where the module developer provides custom UI and server logic for the data transformation.
+But, with the new `teal_transform_module` function, it is now possible to provide a more dynamic transformation capability which can be customised by the app developer and not just the module developer.
+
+The `teal_transform_module()` function can be used to create a `teal_data_module` object that is basically a shiny module where the module server function takes in the data from the filter panel and returns the data after applying the custom transformation logic.
+
+## Creating your first custom data transformation module
+
+We initialize a simple teal app where we pass `iris` and `mtcars` as the input datasets.
+
+```{r, message = FALSE, warning = FALSE}
+library(teal)
+```
+
+```{r}
+data <- teal_data() %>%
+  within({
+    iris <- iris
+    mtcars <- mtcars
+  })
+
+
+app <- init(
+  data = data,
+  modules = example_module()
```

Comment:
Should we use a very simple `module()` so that it's clear that `transformer` is passed as an argument to this function?

Reply:
Yes, I think vignette should elaborate more about creating `module` which consumes transformers.

I've updated description here: